liquibase-integration-tests/src/test/groovy/liquibase/snapshot/JdbcDatabaseConnectionCloseTest.groovy

```diff
@@ -0,0 +1,75 @@
+package liquibase.snapshot
+
+import liquibase.Scope
+import liquibase.database.AbstractJdbcDatabase
+import liquibase.database.Database
+import liquibase.database.jvm.JdbcConnection
+import liquibase.exception.DatabaseException
+import liquibase.extension.testing.testsystem.DatabaseTestSystem
+import liquibase.extension.testing.testsystem.TestSystemFactory
+import liquibase.extension.testing.testsystem.spock.LiquibaseIntegrationTest
+import liquibase.logging.core.BufferedLogService
+import liquibase.parser.ChangeLogParserConfiguration
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.sql.Connection
+import java.util.logging.Level
+
+@LiquibaseIntegrationTest
+class JdbcDatabaseConnectionCloseTest extends Specification {
+    @Shared
+    public DatabaseTestSystem h2 = Scope.currentScope.getSingleton(TestSystemFactory).getTestSystem("h2")
+
+    def "Test Exception"() {
+        when:
+        JdbcConnection connection = new TestJdbcConnection(h2.getConnection())
+        Database database = h2.getDatabaseFromFactory()
+        database.setConnection(connection)
+        database.close()
```

Comment:
why closing the connection will throw an exception? If that's the case can we rename the test to something like "validate exception is thrown when connection is closed"?

Reply:
Yes, thank you for noticing that.